lib/worker.js

```diff
@@ -3,89 +3,52 @@
 process.title = 'linter-eslint helper'
 
 const CP = require('childprocess-promise')
-const ChildProcess = require('child_process')
+const execFileSync = require('child_process').execFileSync
 const Path = require('path')
-const FS = require('fs')
+
 const resolveEnv = require('resolve-env')
+const Helpers = require('./es5-helpers')
 
+const findEslintDir = Helpers.findEslintDir
+const find = Helpers.find
+const determineConfigFile = Helpers.determineConfigFile
+const getEslintCli = Helpers.getEslintCli
 const Communication = new CP()
 
+// closed-over module-scope variables
 let eslintPath = null
-const eslintPathLocal = Path.join(FS.realpathSync(Path.join(__dirname, '..')), 'node_modules', 'eslint')
 let eslint = null
-let prefixPath = null
-
-function find(startDir, name) {
-  let filePath
-  const chunks = startDir.split(Path.sep)
-  while (chunks.length) {
-    filePath = Path.join(chunks.join(Path.sep), name)
-    try {
-      FS.accessSync(filePath, FS.R_OK)
-      return filePath
-    } catch (_) { }
-    chunks.pop()
-  }
-  return null
-}
 
 Communication.on('JOB', function(job) {
   const params = job.Message
+  const modulesPath = find(params.fileDir, 'node_modules')
+  const eslintignoreDir = Path.dirname(find(params.fileDir, '.eslintignore'))
   let configFile = null
   global.__LINTER_RESPONSE = []
 
-  configFile = find(params.fileDir, '.eslintrc')
+  // Check for config file and determine whether to bail out
+  configFile = determineConfigFile(params)
+
   if (params.canDisable && configFile === null) {
     job.Response = []
     return
-  } else if (params.configFile) {
-    configFile = params.configFile
   }
 
-  const modulesPath = find(params.fileDir, 'node_modules')
-  const eslintignoreDir = Path.dirname(find(params.fileDir, '.eslintignore'))
-  let eslintNewPath = null
   if (modulesPath) {
     process.env.NODE_PATH = modulesPath
   } else process.env.NODE_PATH = ''
   require('module').Module._initPaths()
```

Comment:
`process.env.NODE_PATH` change is not recognized by the node module loader, this line makes sure it does